src/components/Buttons/SubmitButton.jsx

```diff
@@ -25,26 +43,40 @@ export const SubmitButton = ({ color = "primary", isLoading, className, children
             <span
                 className={clsx(
                     "absolute inset-0 flex items-center justify-center",
-                    isLoading ? "opacity-100" : "opacity-0",
+                    showTransition ? "opacity-100" : "opacity-0",
                 )}
             >
-                <Transition
-                    appear
-                    as="span"
-                    show={isLoading}
-                    enter="transition-all duration-700"
-                    enterFrom="opacity-0"
-                    enterTo="opacity-100"
-                >
-                    <Spinner size="current" color="current" className="relative -top-0.25 text-white" />
-                </Transition>
+                {showTransition && (
+                    <Transition
+                        appear
+                        as="span"
+                        show={showTransition}
+                        enter="transition-all duration-700"
+                        enterFrom="opacity-0"
+                        enterTo="opacity-100"
+                    >
+                        {showSuccess && (
+                            <CheckIcon
+                                size="medium"
+                                color="current"
+                                className={clsx("relative -top-0.25 text-white", {
+                                    "text-black": rest.variant === "outline",
```

Comment:
Ugh, you never use `rest.` If you know a named arg, then put it in the component. So you will have a `variant` prop above.